src/common/hooks/useBoundPieces/useBoundPieces.js

```diff
@@ -29,8 +46,34 @@ const useBoundPieces = (itemId, options = {}) => {
     refetch,
   } = useQuery({
     queryKey: [namespace, itemId],
-    queryFn: () => ky.get(ORDER_PIECES_API, { searchParams }).json(),
-    enabled: Boolean(enabled && itemId),
+    queryFn: async ({ signal }) => {
+      const makeRequest = (httpClient) => httpClient.get(ORDER_PIECES_API, { searchParams, signal }).json();
+
+      if (isCentralOrderingEnabled) {
+        const fulfilledResults = await Promise.allSettled([
+          makeRequest(ky),
+          makeRequest(extendKyWithTenant(ky, consortiumCentralTenantId))
+        ]).then((results) => results.filter(({ status }) => status === 'fulfilled'));
+
+        const totalRecords = fulfilledResults.reduce((acc, { value }) => acc + value.totalRecords, 0);
+        const pieces = fulfilledResults
+          .flatMap(({ value }, index) => value.pieces.map(piece => ({ ...piece, tenantId: index === 1 ? consortiumCentralTenantId : stripes.okapi.tenant })))
+          .sort((a, b) => new Date(a[SORTING_FIELD]) - new Date(b[SORTING_FIELD]));
+
+        return {
+          pieces,
+          totalRecords,
+        };
+      }
+
+      return makeRequest(ky);
+    },
+    enabled: Boolean(
+      enabled
+      && itemId
+      && consortiumCentralTenantId
+      && !isCentralOrderingSettingsFetching
+    ),
```

Comment:
[stripes-acq-components::getConsortiumCentralTenantId()](https://github.com/folio-org/stripes-acq-components/blob/161e6644c91066a00ca80692894c49decc198079/lib/utils/consortia/getConsortiumCentralTenantId.js#L2) will return `undefined` for non-ECS tenants. Thus, after this change, this hook will only be enabled for ECS tenants. I don't think this is correct.[](url)

Reply:
Good catch, thanks! Will fix it

Reply:
Updated